sympy/solvers/ode.py

```diff
@@ -1187,6 +1188,13 @@ class in it.  Note that a hint may do this anyway if
             # m1[coeff]*m1[x]*m1[y] + m2[coeff]*m2[x]*m2[y]*y'
             m1 = separatevars(r[d], dict=True, symbols=(x, y))
             m2 = separatevars(r[e], dict=True, symbols=(x, y))
+            if not (m1 and m2):
+                try:
+                    m1 = separate_vars(r[d], symbols=(x,y))
```

Comment:
Why are you making a new function instead of fixing/modifying the old one? It's fine to replace the old one with something new that does everything the old one did. Else, add a new step to the old one so it works better.